### PR TITLE
[#73] Add json field names to oblect struct

### PR DESCRIPTION
**Pull request by fyrchik**
Related #73.

### CODE REVIEW

**Comment by alexvanin** [comment]
Seems like protobuf linter does not support json mapping.

**Comment by realloc** [comment]
> Seems like protobuf linter does not support json mapping.

It does. It actually works with AST-like structure. However, when not set explicitly, the JSON mapping names are automatically transformed from filed names. In the check it gives errors because of that field name changes break compatibility.

Think, we can ignore this at initial mapping addition.

File: object/types.proto [hunk 1]
```diff
@@ -32,53 +32,53 @@ enum MatchType {
...
+  uint64 creation_epoch = 4 [json_name = "CreationEpoch"];
 
   // Size of payload in bytes.
   // `0xFFFFFFFFFFFFFFFF` means `payload_length` is unknown
-  uint64 payload_length = 5;
+  uint64 payload_length = 5 [json_name = "PayloadSize"];
 
   // Hash of payload bytes
-  neo.fs.v2.refs.Checksum payload_hash = 6;
+  neo.fs.v2.refs.Checksum payload_hash = 6  [json_name = "PayloadHash"];
```
Comment (alexvanin) [inline object/types.proto:71]: Two spaces between 6 and [

File: object/types.proto [hunk 2]
```diff
@@ -32,53 +32,53 @@ enum MatchType {
...
 
   // Type of the object payload content
-  ObjectType object_type = 7;
+  ObjectType object_type = 7  [json_name = "Type"];
```
Comment (alexvanin) [inline object/types.proto:74]: Two spaces between 7 and [

File: object/types.proto [hunk 3]
```diff
@@ -93,50 +93,50 @@ message Header {
...
   // the original one is in the `Split` headers. Parent and children objects
   // must be within the same container.
   message Split {
     // Identifier of the origin object. Known only to the minor child.
-    neo.fs.v2.refs.ObjectID parent = 1;
+    neo.fs.v2.refs.ObjectID parent = 1 [json_name = "Parent"];
 
     // Identifier of the left split neighbor
-    neo.fs.v2.refs.ObjectID previous = 2;
+    neo.fs.v2.refs.ObjectID previous = 2[json_name = "Previous"];
```
Comment (alexvanin) [inline object/types.proto:112]: No spaces between 2 and [

File: object/types.proto [hunk 4]
```diff
@@ -93,50 +93,50 @@ message Header {
...
 
     // `signature` field of the parent object. Used to reconstruct parent.
-    neo.fs.v2.refs.Signature parent_signature = 3;
+    neo.fs.v2.refs.Signature parent_signature = 3[json_name = "ParentSignature"];
```
Comment (alexvanin) [inline object/types.proto:115]: No spaces between 3 and [

File: object/types.proto [hunk 5]
```diff
@@ -32,53 +32,53 @@ enum MatchType {
...
 // Object Header
 message Header {
   // Object format version. Effectively the version of API library used to
   // create particular object
-  neo.fs.v2.refs.Version version = 1;
+  neo.fs.v2.refs.Version version = 1 [json_name = "Version"];
 
   // Object's container
-  neo.fs.v2.refs.ContainerID container_id = 2;
+  neo.fs.v2.refs.ContainerID container_id = 2 [json_name = "Container"];
```
Comment (alexvanin) [inline object/types.proto:58]: So it'd be like this?
```json
{
    "Container": {
        "Value": "Q29udGFpbmVySUQ="
    },
}
```

Maybe `ContainerID` name will be better there.

**Review by alexvanin — APPROVED** [review]
LGTM, but see the comments.

**Comment by realloc** [comment]
To avoid confusion, I suggest to use the same filed naming as in protobuf definition. For example if the field is named `OwnerID` in protobuf, it should also be named `ownerID` in JSON.

**Review by cthulhu-rider — APPROVED** [review]

**Review by alexvanin — APPROVED** [review]

**Review by im-kulikov — APPROVED** [review]

**Pull request merged**